src/stdio/src/default.rs

```diff
@@ -0,0 +1,85 @@
+use core::sync::atomic::AtomicBool;
+use core::ptr;
+use super::{constants, internal, BUFSIZ, FILE, UNGET};
+
+#[allow(non_upper_case_globals)]
+static mut default_stdin_buf: [u8; BUFSIZ as usize + UNGET] = [0; BUFSIZ as usize + UNGET];
+
+#[allow(non_upper_case_globals)]
+static mut default_stdin: FILE = FILE {
+    flags: constants::F_PERM | constants::F_NOWR | constants::F_BADJ,
+    rpos: ptr::null_mut(),
+    rend: ptr::null_mut(),
+    wend: ptr::null_mut(),
+    wpos: ptr::null_mut(),
+    wbase: ptr::null_mut(),
+    fd: 0,
+    buf: unsafe { &mut default_stdin_buf as *mut [u8] as *mut u8 },
+    buf_size: BUFSIZ as usize,
+    buf_char: -1,
```

Comment:
Instead of using -1 as an uset, we could change `buf_char: Option<u8>`.

Reply:
-1 is being used for EOF